### PR TITLE
feat(button): introduce `size` variant

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Description

Adding small variant to the button component. It's will be used in the future for the download page to have a button kind of less agressive.

## Validation

**Default**
Capture d’écran 2024-12-22 à 11 59 31

**Small**
Capture d’écran 2024-12-22 à 11 59 12

## Related Issues

https://github.com/nodejs/nodejs.org/issues/7037#issuecomment-2558409941

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.

### CODE REVIEW

File: apps/site/components/Common/Button/index.module.css [hunk 1]
```diff
@@ -17,6 +17,12 @@
     @apply cursor-not-allowed;
   }
 
+  &.small {
+    @apply px-3
+      py-2
+      text-sm;
```
Comment (ovflowd) [inline apps/site/components/Common/Button/index.module.css:23]: What would this be in px? 14px?

**Review by ovflowd — APPROVED** [review]
LGTM

File: apps/site/components/Common/Button/index.module.css (hunk 1 above)
Comment (AugustinMauroy) [reply apps/site/components/Common/Button/index.module.css:23]: `px-3` => `12px`
`py-2` => `8px`

```css
.text-sm {
    font-size: 0.875rem /* 14px */;
    line-height: 1.25rem /* 20px */;
}
```

**Review by canerakdas — APPROVED** [review]
LGTM!

File: apps/site/components/Common/Button/index.stories.tsx
Comment (bmuenzenmeyer) [inline apps/site/components/Common/Button/index.stories.tsx:1]: its kinda ironic to me that the defaults like `size` and `disabled` need to be defined here.

**Review by bmuenzenmeyer — APPROVED** [review]

File: apps/site/components/Common/Button/index.stories.tsx
Comment (AugustinMauroy) [reply apps/site/components/Common/Button/index.stories.tsx:1]: yeah but this "issue" from storybook. they need to know props to control them

**Pull request merged**